apps/web/components/ProductProperties/ProductProperties.vue

```diff
@@ -1,50 +1,52 @@
-          @update:model-value="value !== selectedSize && setAttribute('size', value)"
-        >
-          {{ label }}
-        </SfChip>
-      </span>
-    </fieldset>
-    <fieldset v-if="sizes?.length" class="pb-2">
-      <legend class="block mb-2 text-base font-medium leading-6 text-neutral-900">{{ $t('color') }}</legend>
-      <span v-for="{ label, value } in colors" class="mr-2 mb-2 uppercase" :key="value">
-        <SfChip
-          class="min-w-[48px]"
-          size="sm"
-          :input-props="{
-            onClick: (e) => value == selectedSize && e.preventDefault(),
-          }"
-          :model-value="value === selectedColor"
-          @update:model-value="setAttribute('color', value)"
-        >
-          <template #prefix><SfThumbnail size="sm" :style="{ background: value }" /></template>
-          {{ label }}
-        </SfChip>
-      </span>
-    </fieldset>
+  <div v-if="attributeGroups && attributeGroups.length" class="px-4" data-testid="product-properties">
+    <SfSelect class="mb-2" v-for="(attributeGroup, index) in attributeGroups" :key="index"
+      @update:modelValue="changeVariationId" v-model="selectedVariation" size="sm" placeholder="-- Select --">
+      <option v-for="{ value, label } in attributeGroup.map(x => ({ value: x.variationId.toString(), label: x.value }))" :key="value"
+        :value="value">
+        {{ label }}
+      </option>
+    </SfSelect>
   </div>
 </template>
 
 <script setup lang="ts">
-import { SfChip, SfThumbnail } from '@storefront-ui/vue';
+import { productGetters } from '@plentymarkets/plentymarkets-sdk/packages/sdk/src';
+import { SfSelect } from '@storefront-ui/vue';
 import { ProductPropertiesProps } from '~/components/ProductProperties/types';
-
+import { RouteRecordName, RouteParamValueRaw } from 'vue-router';
 const props = defineProps<ProductPropertiesProps>();
-const { getAttributeList, getAttribute, setAttribute } = useProductAttribute(props.product, ['color', 'size']);
+const product = props.product;
+
+interface ProductAttributeInterface {
+  type: string;
+  id: string;
+  value: string;
+  label: string;
+  variationId: number;
+};
+
+
+const attributeGroups = computed((): ProductAttributeInterface[][] => {
+  const groupsForProducts = productGetters.getAttributes([product], ['color', 'size']) as ProductAttributeInterface[][][];
+  if (groupsForProducts) {
+    return groupsForProducts[0]
+  } else {
+    return []
+  }
+}
+);
+
+const router = useRouter();
+const route = useRoute();
+const slug = route.params.slug as string;
+const productId = slug.split('-').pop() ?? '0';
+const selectedVariation = ref(productId);
+
+const changeVariationId = (newId: String) => {
+  const delimiter = '-' as string;
+  const newLink = slug.substring(0, slug.lastIndexOf(delimiter)) + delimiter + `${newId}` as RouteParamValueRaw | (string | number)[];
+  router.push({ name: route.name as RouteRecordName | undefined, params: { slug: newLink } });
```

Comment:
do we really need to type cast like this ?

**as RouteRecordName | undefined**

can the route name ever be undefined ?